src/lpython/utils.cpp

```diff
@@ -102,6 +102,68 @@ std::string get_runtime_library_header_dir()
     return path;
 }
 
+std::string exec_command(const std::string& command) {
+    std::string output;
+    output.resize(1024);
+    std::memset(output.data(), 0, 1024);
+    FILE* fp = popen(command.c_str(), "r");
+    if (fp != nullptr) {
+        std::fgets(output.data(), 1024, fp);
+        std::string result(output.data());
+        result.erase(result.find_last_not_of("\n") + 1);
+        pclose(fp);
+        return result;
+    }
+    return "";
+}
+
+std::string get_python_site_packages() {
+    std::vector<std::string> paths;
+
+    // Try using the site module to find the site-packages directory
+    std::string command = "python -c \"import site; print(site.getsitepackages()[0])\"";
+    std::string path = exec_command(command);
+    if (!path.empty()) {
+        paths.push_back(path);
+    }
+
+    // If the above method didn't work, try using the sys module
+    if (paths.empty()) {
+        command = "python -c \"import sys; print(sys.exec_prefix + '/lib/python' + sys.version[:3] + '/site-packages')\"";
+        path = exec_command(command);
+        if (!path.empty()) {
+            paths.push_back(path);
+        }
+    }
+
+    // If none of the above methods worked, try a few common installation locations
+    if (paths.empty()) {
+        std::string path_prefix;
+        #ifdef _WIN32
+            path_prefix = std::string(std::getenv("APPDATA")) + "/Python";
+        #else
+            path_prefix = "/usr/local/lib/python";
+        #endif
+        for (int major_version = 2; major_version <= 3; major_version++) {
+            for (int minor_version = 0; minor_version <= 9; minor_version++) {
+                std::string path = path_prefix + std::to_string(major_version) + "." + std::to_string(minor_version) + "/site-packages";
+                if (access(path.c_str(), F_OK) != -1) {
+                    paths.push_back(path);
+                }
+            }
+        }
+    }
+
+    // If no site-packages directory was found, return an empty string
+    if (paths.empty()) {
+        return "";
+    }
+
+    // Return the first site-packages directory found
+    return paths[0];
+}
```

Comment:
This seems very fragile.

Why not for now just require the user to supply the correct `-I` flags?